### PR TITLE
Pull request acceptance Checklist

### LINKED ISSUE #293: Add Pull Request Acceptance Check List

### PR DESCRIPTION
**Pull request by acBerger**
fixes #293

### CODE REVIEW

File: doc/Pull Request Acceptance Checklist.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
+# Pull Request Acceptance Checklist
+
+## Beagle Project
+ * detailed expanatory Javadoc
```
Comment (jGleitz) [inline doc/Pull Request Acceptance Checklist.md]: exp**[l]**anatory

File: doc/Pull Request Acceptance Checklist.md [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+ * No checkstyle violations (errors or warnings)
+ * 100% Code coverage by JUnit Tests
+ * JUnit test check were created by another person and are complete (check everything possible)
```
Comment (jGleitz) [inline doc/Pull Request Acceptance Checklist.md]: > (check everything possible)

well that’s impossible and quite vague. Can we thing of a more precise definition of when a PR is sufficiently tested? Is that even possible?

File: doc/Pull Request Acceptance Checklist.md [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+ * Travis 'goes green'. This shows code coverage and checkstyle is okay.
+
+## Prototypes
+ * detailed expanatory Javadoc
```
Comment (jGleitz) [inline doc/Pull Request Acceptance Checklist.md]: exp**[l]**anatory

File: doc/Pull Request Acceptance Checklist.md [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+ * configured Checkstyle, Formatter and Clean Up
+ * No Checkstyle violations (errors or warnings)
+ * Push destination is not our master branch
```
Comment (jGleitz) [inline doc/Pull Request Acceptance Checklist.md]: We removed that requirement: Prototypes are pushed into master

**Comment by jGleitz** [comment]
We could write down our “min 2 lgtm” rule here.

File: doc/Pull Request Acceptance Checklist.md (hunk 2 above)
Comment (acBerger) [inline doc/Pull Request Acceptance Checklist.md]: I do not understand why you think it is impossible to check everything one can check.

File: doc/Pull Request Acceptance Checklist.md [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+ * Travis 'goes green'. This shows code coverage and checkstyle is okay.
+ * discussions about PR are finished
+ * found errors and mistakes are fixed
+ * minimum two 'lgtm'
+
+## Prototypes
+ * detailed explanatory Javadoc
+ * configured Checkstyle, Formatter and Clean Up
+ * No Checkstyle violations (errors or warnings)
+ * Push destination is not our master branch
```
Comment (RomanLangrehr) [inline doc/Pull Request Acceptance Checklist.md]: This is wrond, they should be pushed into master.

File: doc/Pull Request Acceptance Checklist.md (hunk 2 above)
Comment (jGleitz) [inline doc/Pull Request Acceptance Checklist.md]: Well, what do you mean by “check everything one can check”, if a method takes a `long`, do you want to check all 2^64 possible input values?

Comment (acBerger) [inline doc/Pull Request Acceptance Checklist.md]: Okay, I see.
Are you able to describe what is meant, so we can replace this misunderstanding sentence?

Comment (jGleitz) [inline doc/Pull Request Acceptance Checklist.md]: Hard to say. I doubt there is a waterproof definition. Maybe we should just state that "no reviewer can think of any additional, meaningful test case". 

Am 23.12.2015 7:39 nachm. schrieb acBerger <notifications@github.com>:

In doc/Pull Request Acceptance Checklist.md:

> @@ -0,0 +1,15 @@ > +# Pull Request Acceptance Checklist > + > +## Beagle Project > + * detailed expanatory Javadoc > + * No checkstyle violations (errors or warnings) > + * 100% Code coverage by JUnit Tests > + * JUnit test check were created by another person and are complete (check everything possible) 

Okay, I see.
Are you able to describe what is meant, so we can replace this misunderstanding sentence?

**Comment by RomanLangrehr** [comment]
Lgtm

**Comment by jGleitz** [comment]
Lgtm

**Comment by jGleitz** [comment]
Lgtm

**Pull request merged**